server/api/user/user.controller.js

```diff
@@ -320,6 +354,32 @@ export function changePassword(req, res, next) {
 }
 
 /**
+ * Resets a user's password if they forgot it.
+ * Uses email verificaiton via a token
+ */
+export function resetPassword(req, res, next) {
+  var newPass = String(req.body.newPassword);
+  var token = String(req.body.token);
+  User.findOne({ 'verificationToken' : token })
+    .select('_id email password provider salt verificationToken')
```

Comment:
are you sending all of this info to the user? Why would we ever send the password?